lib/saluki-components/src/sources/checks/aggregator.rs

```diff
@@ -1,12 +1,131 @@
 use super::*;
 use pyo3::prelude::*;
+use saluki_event::{metric::*, Event};
+use saluki_env::time::get_unix_timestamp;
 
+#[derive(Clone, Copy)]
+enum PyMetricType {
+    Gauge = 0,
+    Rate,
+    Count,
+    MonotonicCount,
+    Counter,
+    Histogram,
+    Historate,
+}
+
+#[derive(Debug, Snafu)]
+#[snafu(context(suffix(false)))]
+pub enum AggregatorError {
+    UnsupportedType{},
+}
+
+impl TryFrom<i32> for PyMetricType {
+    type Error = ();
+
+    fn try_from(v: i32) -> Result<Self, Self::Error> {
+        match v {
+            x if x == PyMetricType::Gauge as i32 => Ok(PyMetricType::Gauge),
+            x if x == PyMetricType::Rate as i32 => Ok(PyMetricType::Rate),
+            x if x == PyMetricType::Count as i32 => Ok(PyMetricType::Count),
+            x if x == PyMetricType::MonotonicCount as i32 => Ok(PyMetricType::MonotonicCount),
+            x if x == PyMetricType::Counter as i32 => Ok(PyMetricType::Counter),
+            x if x == PyMetricType::Histogram as i32 => Ok(PyMetricType::Histogram),
+            x if x == PyMetricType::Historate as i32 => Ok(PyMetricType::Historate),
+            _ => Err(()),
+        }
+    }
+}
+
+/// CheckMetric are used to transmit metrics from python check execution results
+/// to forward in the saluki's pipeline.
+pub struct CheckMetric {
+    name: String,
+    metric_type: PyMetricType,
+    value: f64,
+    tags: Vec<String>,
+}
+
+// TODO(remy): use TryFrom instead
+pub fn check_metric_as_event(metric: CheckMetric) -> Result<Event, AggregatorError> {
+    let mut tags = MetricTags::default();
+    // TODO(remy): do this more idiomatically
+    for tag in metric.tags.iter() {
+        tags.insert_tag(tag.clone());
+    }
+
+    let context = MetricContext{
+                    name: metric.name,
+                    tags,
+    };
+    let metadata = MetricMetadata::from_timestamp(get_unix_timestamp());
+
+    match metric.metric_type {
+        PyMetricType::Gauge => {
+            Ok(saluki_event::Event::Metric(Metric::from_parts(
+                context, MetricValue::Gauge { value: metric.value, }, metadata,
+            )))
+        },
+        PyMetricType::Counter => {
+            Ok(saluki_event::Event::Metric(Metric::from_parts(
+                context, MetricValue::Counter { value: metric.value, }, metadata,
+            )))
+        },
+        // TODO(remy): rest of the types
+        _ => Err(AggregatorError::UnsupportedType{}),
+    }
+}
+
+impl Clone for CheckMetric {
+    fn clone(&self) -> Self {
+        Self {
+            name: self.name.clone(),
+            metric_type: self.metric_type,
+            value: self.value.clone(),
+            tags: self.tags.clone(),
+        }
+    }
+}
+
+
+/// Global for Python checks execution to report the data
+pub static SUBMISSION_QUEUE: Lazy<Mutex<Queue<CheckMetric>>> = Lazy::new(|| {
+    Mutex::new(queue![])
+});
```

Comment:
This seems like it could easily end up leading to multiple checks having their metrics interleaved in the queue. Is there any requirement that after triggering a check, we need to be able to extract only their specific metrics?